llvm/lib/TargetParser/RISCVISAInfo.cpp

```diff
@@ -116,6 +115,44 @@ void llvm::riscvExtensionsHelp(StringMap<StringRef> DescMap) {
             "For example, clang -march=rv32i_v1p0\n";
 }
 
+void RISCVISAInfo::printEnabledExtensions(
+    bool IsRV64, std::set<StringRef> &EnabledFeatureNames,
+    StringMap<StringRef> &DescMap) {
+  outs() << "Extensions enabled for the given RISC-V target\n\n";
+  PrintExtension("Name", "Version", (DescMap.empty() ? "" : "Description"));
+
+  RISCVISAUtils::OrderedExtensionMap FullExtMap;
+  RISCVISAUtils::OrderedExtensionMap ExtMap;
+  for (const auto &E : SupportedExtensions)
+    if (EnabledFeatureNames.count(E.Name) != 0) {
+      FullExtMap[E.Name] = {E.Version.Major, E.Version.Minor};
+      ExtMap[E.Name] = {E.Version.Major, E.Version.Minor};
+    }
+  for (const auto &E : ExtMap) {
+    std::string Version =
+        std::to_string(E.second.Major) + "." + std::to_string(E.second.Minor);
+    PrintExtension(E.first, Version, DescMap[E.first]);
+  }
```

Comment:
```suggestion
  RISCVISAUtils::OrderedExtensionMap FullExtMap;
  for (const auto &E : SupportedExtensions)
    if (EnabledFeatureNames.count(E.Name) != 0) {
      FullExtMap[E.Name] = {E.Version.Major, E.Version.Minor};
       std::string Version =
        std::to_string(E.Version.Major) + "." + std::to_string(E.Version.Minor);
        PrintExtension(E.Name, Version, DescMap[E.Name]);
    }
```
`SupportedExtensions` is sorted.

Reply:
It's sorted but not in the same order as OrderedExtensionMap.

Reply:
I propose that we print with the current order proposed in this patch (canonical). If we decide that we'd like to change printing order, then we should have a follow up PR that changes print order for both `printSupportedExtensions` and `printEnabledExtensions`.